### PR TITLE
fix(core-library): do not create pr when a pr already exists for a ref

### LINKED ISSUE #17: Framework-core: handle existing PRs and existing branches
- [x] When there is an existing PR on an up-stream repository and a PR from the same branch and same down-stream repository is opened, there will be an error thrown. Ensure that the latest version is made into a PR.
- [x] When there is an existing branch and someone tries to apply changes onto an existing branch, optionally overwrite the existing branch. If overwrite is enabled, overwrite. Otherwise gracefully fail.

### Additional Information
This is for branches on a fork from an upstream-repository, and PRs from a fork on an upstream-repository

### PR DESCRIPTION
**Pull request by TomKristie**
Recall that GitHub PRs just point to a reference head which are applied on-top of some base reference head, typically master, and so when the reference gets updated the PR automatically gets updated as well.

There may be scenarios where a pull request is already created for a ref, and a user will re-invoke code-suggester create pr using the same ref. If the `force` flag/parameter is true, a new ref is not created. Instead code-suggester create pr will force-update the existing reference. If we try to re-create the a pr for the same reference, there will be a duplication issue. Hence, if there is an existing pr, just update the ref, and GitHub will just adjust UI display accordingly to the reference HEAD data. However, if `force` flag/parameter is false, code-suggester create pr will fail when trying to create the reference because there is a duplicate reference.

**TL;DR**
When there is an existing pr for the a reference and code-suggester create pr is re-invoked, do not create a new pr. Instead, re-use the old one and update the ref.

Closes #17

### CODE REVIEW

**Comment by TomKristie** [comment]
> So this handles the case when essentially there is an existing branch. Is this not the same case as the force option?

Correct. Basically if there is an existing PR for the branch, do not create a new PR, re-use the old one. GitHub PRs just point to a reference head which are applied on-top of some reference head, typically master.

File: test/pr.ts [hunk 1]
```diff
@@ -23,60 +23,287 @@ before(() => {
...
+      repo: similarOrigin1.repo,
+      title: description.title,
+      head: 'octocat:new-topic-1',
+      base: 'master',
+      body: description.body,
+      maintainer_can_modify: true,
+    });
+  });
+
+  it('Invokes octokit pull create when there are similar refs with pull requests open, but not exact refs: extra number', async () => {
```
Comment (chingor13) [inline test/pr.ts:118]: You can nest describe blocks - many of these tests share the same setup for the case of an already existing PR.

```typescript
describe('when there are similar refs with pull requests open', async () => {
  beforeEach(() => {
    // set up sinon for list pull request
  });
  it('should create another pull request if it's not the exact ref', async () => {

  });
});
```

File: test/pr.ts [hunk 2]
```diff
@@ -23,60 +23,239 @@ before(() => {
...
+
+  describe('When there are similar refs with pull requests open, the current new and unique ref still opens a pr', async () => {
+    const responseCreatePullData = await import(
+      './fixtures/create-pr-response.json'
+    );
+    const responseListPullData = await import(
+      './fixtures/list-pulls-response.json'
+    );
+    afterEach(() => {
+      sandbox.restore();
```
Comment (chingor13) [inline test/pr.ts:83]: This should be handled by the outer describe blocks's `afterEach` which is restoring the sandbox.
